Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp

```diff
@@ -171,11 +160,47 @@ void UPSWorldSubsystem::OnWorldBeginPlay(UWorld& InWorld)
 void UPSWorldSubsystem::Deinitialize()
 {
 	Super::Deinitialize();
-	PSHUDComponentInternal = nullptr;
-	PSCurrentSpotComponentInternal = nullptr;
-	SaveGameDataInternal = nullptr;
-	ProgressionSettingsDataInternal.Empty();
-	StarDynamicProgressMaterial = nullptr;
+}
+
+// Invoked after a game feature plugin is unloaded
+void UPSWorldSubsystem::OnGameFeatureUnloading(const UGameFeatureData* GameFeatureData, const FString& PluginURL)
+{
+	DestroyStarActors();
```

Comment:
- Call `DestroyStarActors()` if only it's Progression System. Right now you will destroy stars even if Menu module unloaded. If you unload and load menu module: you will notice stars destroyed as well. It's a bug
- Right after DestroyStarActors, on next line also remove Observer from MGF system

Reply:
done: https://github.com/h4rdmol/ProgressionSystem/pull/26/commits/ffb6191a2d4d2ab08b696a5a8872a8d922016f94